### ISSUE TITLE
Migrate to AWS v2 SDK

### ISSUE THREAD

**Issue by MovieStoreGuy** [open-telemetry/opentelemetry-collector-contrib#36699]
## Context

AWS has annouced that it will be discontinuing support for go AWS SDK v1 and we should be updating the collector to migrate away from it to ensure any issues with the package are dealt with.

https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-aws-sdk-for-go-v1-on-july-31-2025/

**Comment by MovieStoreGuy** [comment]
Some of these may already be migrated off, just confirming as we go.

**Comment by MovieStoreGuy** [comment]
For those wanting to pick up this issue,

The migration guides https://aws.github.io/aws-sdk-go-v2/docs/migrating/ is a good starting place to understand what needs to be done.

**Comment by LZiHaN** [comment]
hi @MovieStoreGuy , is this issue still available? i'd like to working on it, assign please

**Comment by mx-psi** [comment]
Duplicate of #33208?

**Comment by MovieStoreGuy** [comment]
> Duplicate of #33208?

Yes, sorry

**Comment by crobert-1** [comment]
_I've checked off `confmap/provider` as being complete, all AWS references are v2_

**Comment by crobert-1** [comment]
_I've added links to open and merged PRs to the issue description, hopefully it's helpful. Feel free to reformat as necessary._

**Comment by crobert-1** [comment]
_I've opened sub-issues for all of the components that don't have an open PR yet. Depending on PR outcomes I can create issues for more components as well._

**Comment by mjlshen** [comment]
@crobert-1 When you get a chance, could you help create a sub-issue for [`extension/awsproxy`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/a00b4c92513b1b23f30bf90d715489d44da95eb6/extension/awsproxy/go.mod#L22)? While reviewing the last couple PRs I just realized this was still outstanding. I'm happy to work on it in the meantime, it seems that I can't create sub-issues myself.

**Comment by crobert-1** [comment]
> [@crobert-1](https://github.com/crobert-1) When you get a chance, could you help create a sub-issue for [`extension/awsproxy`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/a00b4c92513b1b23f30bf90d715489d44da95eb6/extension/awsproxy/go.mod#L22)? While reviewing the last couple PRs I just realized this was still outstanding. I'm happy to work on it in the meantime, it seems that I can't create sub-issues myself.

Thanks for letting me know @mjlshen! I've opened https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/40461 👍

**Comment by rgoltz** [comment]
Still relevant, since some components remains open. (update for bot)

**Comment by anarwal** [comment]
Hi @rgoltz 

I’ve been going through the AWS SDK v1 → v2 migration effort and reviewing the current state across sub-issues.

It looks like most components have been partially migrated, but some areas (e.g., internal/aws, especially proxy-related code) may still have remaining SDK v1 usage. I’ve started looking into #37728 to understand the current gaps and identify what’s left to complete.

While I am waiting on maintainers for that component, I wanted to check here:
- Is there any preferred prioritization across remaining components?
- Would it make sense to focus on completing internal/aws (e.g., proxy and awsutil) first?

My plan is to:
- identify remaining SDK v1 usages in internal/aws
- focus on completing migration for a specific sub-component
- and help unblock any pending work if needed

Happy to align with any guidance on where help is most needed.

**Comment by mx-psi** [comment]
@anarwal ThanI think it would be worth adding a `depguard` rule [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/08e8d9ed1d84e61025da664e1a5d7f6a96618e1f/.golangci.yml#L187) for the v1 version with exceptions for each component that has not migrated yet as a first step

**Comment by anarwal** [comment]
@mx-psi That makes sense - I think adding a `depguard` rule sounds good as it will prevent any new SDK v1 usage while the migration is still in progress.

I’ll take a look at the current golangci config and work on:
- adding a depguard rule to block AWS SDK v1
- identifying components that still require exceptions (based on current usage)
- scoping the rule so it doesn’t break existing builds while we complete the migration

I’ll start with a draft PR over the weekend and share it for feedback.

**Comment by anarwal** [comment]
@mx-psi I've opened a PR to add the `depguard` rule as discussed: #48647 
                                                                                                                                                                                                                                                                                                                
This adds a `depguard` rule blocking github.com/aws/aws-sdk-go, for now I have not added any excpeitons but some of them might be need (I was working on #37728, which needs):

```
allow:
- exporter/prometheusexporter
- extension/asapauthextension
- internal/datadog/e2e
- receiver/prometheusreceiver
- receiver/simpleprometheusreceiver
- receiver/purefbreceiver
- receiver/purefareceiver
- testbed
```

Please let me know if we should be adding these exceptions or not.

**Comment by anarwal** [comment]
@mx-psi I checked the original list of components in the [issue ](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36699#issue-2721798380) and all have been migrated to AWS SDK v2. Verified no remaining v1 usage exists:                                                                                                                                                                                          
 
```                                                                                                                                                                                                                                                                                                               
rg 'github.com/aws/aws-sdk-go' -l -tgo | cut -d'/' -f1-2 | sort | uniq
```                                                                                                                                                                                                                                                                                                        

This returned no results, confirming all v1 imports have been removed.                                                                                                                                                                                                                                        
                                                                                                                                                                                                                                                                                                                
To prevent future regressions, as you suggested I have added `depguard` lint rule to block new github.com/aws/aws-sdk-go (v1) imports across the repository (PR pending merge). Any new code importing the v1 SDK will now fail CI with a message pointing to v2. This should fully address the migration effort and we can close this issue.
